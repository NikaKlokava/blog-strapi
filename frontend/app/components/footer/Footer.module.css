.footer_container {
  display: flex;
  justify-content: space-between;
  padding: 35px;
  align-items: center;
}

.links_container {
  display: flex;
  column-gap: 10px;
}

.icon {
  width: 30px;
  height: 30px;
  background-size: 30px 30px;
  opacity: 0.7;
  cursor: pointer;
}


@media screen and (max-width: 500px) {
  .footer_container {
    flex-direction: column;
    padding: 15px;
    row-gap: 15px;
  }
}