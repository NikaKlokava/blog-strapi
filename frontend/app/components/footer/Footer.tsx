import styles from "./Footer.module.css";
import Image from "next/image";

type Data = {
  id: string;
  attributes: {
    name: string;
    href: string;
    createdAt: string;
    updatedAt: string;
    publishedAt: string;
    svg: any;
  };
}[];
export const Footer = async () => {
  const data: Data = await fetchData();

  return (
    <div className={styles.footer_container}>
      <p className="text-sm">Created by Veranika Klokava</p>
      <p className="text-sm">2023</p>
      <div className={styles.links_container}>
        {data.map((link) => {
          return (
            <a
              key={link.attributes.name}
              className={styles.icon}
              href={`${link.attributes.href}`}
            >
              <Image
                src={
                  process.env.STRAPI_API_URL +
                  link.attributes.svg.data.attributes.url
                }
                width={30}
                height={30}
                alt={`${link.attributes.name}_svg`}
              />
            </a>
          );
        })}
      </div>
    </div>
  );
};

const fetchData = async () => {
  const response = await fetch(
    `${process.env.STRAPI_API_URL}/api/links?populate=*`
  );
  const result = await response.json();
  return result.data;
};
