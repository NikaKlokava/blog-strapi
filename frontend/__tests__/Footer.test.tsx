import { Footer } from "@/app/components/footer/Footer";
import { render } from "@testing-library/react";

describe("Test the Footer component", () => {
  it("The Footer component renders correctly", async () => {
    const footerSnap = render(await (async () => await Footer())());
    expect(footerSnap).toMatchSnapshot();
  });
})

