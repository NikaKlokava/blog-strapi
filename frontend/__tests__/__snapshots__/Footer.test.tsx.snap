// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Footer component The Footer component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer_container"
      >
        <p
          class="text-sm"
        >
          Created by Veranika Klokava
        </p>
        <p
          class="text-sm"
        >
          2023
        </p>
        <div
          class="links_container"
        >
          <a
            class="icon"
            href="https://www.linkedin.com/in/veranika-klokava-858b5b287"
          >
            <img
              alt="linkedin_svg"
              data-nimg="1"
              decoding="async"
              height="30"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="30"
            />
          </a>
          <a
            class="icon"
            href="https://github.com/NikaKlokava"
          >
            <img
              alt="github_svg"
              data-nimg="1"
              decoding="async"
              height="30"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="30"
            />
          </a>
          <a
            class="icon"
            href="https://www.instagram.com/nika_klokava"
          >
            <img
              alt="instagram_svg"
              data-nimg="1"
              decoding="async"
              height="30"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="30"
            />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer_container"
    >
      <p
        class="text-sm"
      >
        Created by Veranika Klokava
      </p>
      <p
        class="text-sm"
      >
        2023
      </p>
      <div
        class="links_container"
      >
        <a
          class="icon"
          href="https://www.linkedin.com/in/veranika-klokava-858b5b287"
        >
          <img
            alt="linkedin_svg"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="30"
          />
        </a>
        <a
          class="icon"
          href="https://github.com/NikaKlokava"
        >
          <img
            alt="github_svg"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="30"
          />
        </a>
        <a
          class="icon"
          href="https://www.instagram.com/nika_klokava"
        >
          <img
            alt="instagram_svg"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="30"
          />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
